### ISSUE TITLE
trend feature for GBM(s)

### ISSUE THREAD

**Issue by Loudegaste** [unit8co/darts#1760]
**Problem:**
GBMs are, in a lot of areas of ML, the most robust/efficient/easy to use ML algorithm and it's great to have them in Darts.
However, they cannot extrapolate beyond the range of values from training set. Timeseries often have trends (linear, poly or exp) which will not be interacting with the other components of the TS but will cause the test set values to be outside of the training range, thus destroying the performance. 

**Solution:**
This could be solved fairely simply by adding a trend parameter e.g(trend=None/poly1/pol2/exp) parameter which:
1. fits the trend
2. detrends the TS
3. pass the detrended TS to the underlying GBM model
4. Retrend the output from the GBM model

**Comment by dennisbader** [comment]
As discussed today, we rather stick with how LGBM is currently implemented.
The trend/seasonality decomposition can be applied as a pre processing step before modelling with LGBM.

- `TimeSeries.diff()`, `Diff` transformer: remove trend from time series
- model trend / seasonality with separate model and remove it from original time series
